examples/ecies-example.cpp

```diff
@@ -0,0 +1,265 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+#include <mococrw/ecies.h>
+#include <mococrw/hash.h>
+#include <mococrw/util.h>
+#include <mococrw/openssl_wrap.h>
+
+#include <iostream>
+
+using namespace mococrw;
+
+struct EciesEncryptResult {
+    std::vector<uint8_t> ephemeralKey;
+    std::vector<uint8_t> ciphertext;
+    std::vector<uint8_t> mac;
+};
+
+EciesEncryptResult encrypt_data(const std::vector<uint8_t> &message)
+{
+    const std::string pubKeyPem = R"(
+-----BEGIN PUBLIC KEY-----
+MHYwEAYHKoZIzj0CAQYFK4EEACIDYgAELGj/fqXab+amxDOcbDWMArPBsUMOPons
+9NePVyS9CbIkI8e3nPYi3ytHjJm03M22vM5R4XAxI9cMv5biylFJW0HBlyf/cZTO
+xOVCTvRDUHLGTdGXNlz74YtWLF+CMX5A
+-----END PUBLIC KEY-----
+    )";
+
+    /* The standard for ECIES (IEEE 1363a-2004) doesn't specify the details of the different cipher blocks.
+     * Thus it is up to the implementer to define its own protocol (hash-function, key derivation function,
+     * message authentication code, symmetric cipher, key serialization) used for encryption and decryption.
+     * In the current example the following is used:
+     * - SHA512 everywhere where a hash function is required
+     * - X963(SHA512) as key derivation function
+     * - HMAC(SHA512) as message authentication code
+     * - AES CBC with PKCS padding, zero IV and 256 bit key size
+     * - Additionally salts for KDF and MAC can be set (optional value, default: empty)
+     *
+     * The defaults if the builder is invoked without any optional parameters are:
+     * - X963(SHA512) as key derivation function
+     * - HMAC(SHA512)
+     * - AES CBC with PKCS padding, zero IV and 256 bit key size
+     * - empty salts for KDF and MAC
+     */
+    /* Read the public key */
+    auto pubKey = AsymmetricPublicKey::readPublicKeyFromPEM(pubKeyPem);
+
+    /* Get the encryption context */
+    ECIESCtxBuilder encBuilder;
+    std::unique_ptr<ECIESEncryptionCtx> encCtx = encBuilder
```

Comment:
```suggestion
    std::unique_ptr<ECIESEncryptionCtx> encCtx = ECIESCtxBuilder{}
```

Could we create an anonymous temporary ECIESCtxBuilder object here, rather than explicitly assigning it to a variable?